web/packages/shared/components/UnifiedResources/ListView/ResourceListItem.tsx

```diff
@@ -0,0 +1,367 @@
+      ) {
+        setIsNameOverflowed(true);
+      } else {
+        setIsNameOverflowed(false);
+      }
+      if (
+        descText.current?.scrollWidth >
+        descText.current?.parentElement.offsetWidth
+      ) {
+        setIsDescOverflowed(true);
+      } else {
+        setIsDescOverflowed(false);
+      }
+    });
+
+    observer.observe(innerContainer.current);
+    return () => {
+      observer.disconnect();
+    };
+  });
+
+  return (
+    <RowContainer
+      onMouseEnter={() => setHovered(true)}
+      onMouseLeave={() => setHovered(false)}
+    >
+      <RowInnerContainer
+        ref={innerContainer}
+        alignItems="start"
+        pinned={pinned}
+        selected={selected}
+      >
+        {/* checkbox */}
+        <HoverTooltip
+          css={`
+            grid-area: checkbox;
+          `}
+          tipContent={<>{selected ? 'Deselect' : 'Select'}</>}
+        >
+          <StyledCheckbox checked={selected} onChange={selectResource} />
+        </HoverTooltip>
+
+        {/* pin button */}
+        {/* We wrap it in a box in order to center it properly in its grid. */}
+        <Box
+          css={`
+            grid-area: pin;
+            place-self: center center;
+          `}
+        >
+          <PinButton
+            setPinned={pinResource}
+            pinned={pinned}
+            pinningSupport={pinningSupport}
+            hovered={hovered}
+          />
+        </Box>
+
+        {/* icon */}
+        <ResourceIcon
+          name={primaryIconName}
+          width="36px"
+          height="36px"
+          css={`
+            grid-area: icon;
+            place-self: center center;
+          `}
+        />
+
+        {/* name and description */}
+        <Flex
+          css={`
+            grid-area: name;
+            justify-content: left;
+            align-items: center;
+            overflow: hidden;
+          `}
+        >
+          <Flex
+            flexDirection="column"
+            css={`
+              overflow: hidden;
+            `}
+          >
+            {isNameOverflowed ? (
+              <HoverTooltip tipContent={<>{name}</>}>
+                <Name ref={nameText}>{name}</Name>
+              </HoverTooltip>
+            ) : (
+              <Name ref={nameText}>{name}</Name>
+            )}
+            {description && (
+              <>
+                {isDescOverflowed ? (
+                  <HoverTooltip tipContent={<>{description}</>}>
+                    <Description ref={descText}>{description}</Description>
+                  </HoverTooltip>
+                ) : (
+                  <Description ref={descText}>{description}</Description>
+                )}
+              </>
+            )}
+          </Flex>
+          <Box
+            css={`
+              align-self: start;
+            `}
+          >
+            {hovered && <CopyButton name={name} />}
+          </Box>
+        </Flex>
+
+        {/* type */}
```

Comment:
When the viewport is narrow, the resource type looks like this:
<img width="723" alt="image" src="https://github.com/gravitational/teleport/assets/20583051/b1204bea-de95-4f37-a60d-551319234d52">

I see `ellipsis` on the container so I guess it was supposed to behave differently.